tests/pass-dep/tokio/sleep.rs

```diff
@@ -1,14 +1,17 @@
-//@compile-flags: -Zmiri-disable-isolation -Zmiri-permissive-provenance -Zmiri-backtrace=full
+//@compile-flags: -Zmiri-permissive-provenance -Zmiri-backtrace=full
 //@only-target-x86_64-unknown-linux: support for tokio only on linux and x86
 
 use tokio::time::{sleep, Duration, Instant};
 
 #[tokio::main]
 async fn main() {
+    // We are using a very short time for sleep, 1 millisecond, (and ensure sleep waits for at
+    // least 1000 microseconds, the same as 1 millisecond) because this test is using miri's
+    // virtual clock due to isolation. The virtual clock works by using 1 code block passed as 10
+    // nanoseconds. While writing this test, I saw 1,000,000 basic blocks had been executed pretty
+    // quickly which would be the equivalent of 1 millisecond.
     let start = Instant::now();
-    sleep(Duration::from_secs(1)).await;
-    // It takes 96 millisecond to sleep for 1 millisecond
-    // It takes 1025 millisecond to sleep for 1 second
-    let time_elapsed = &start.elapsed().as_millis();
-    assert!(time_elapsed > &1000, "{}", time_elapsed);
+    sleep(Duration::from_millis(1)).await;
```

Comment:
Which underlying sleeping primitive does tokio even use here?

Reply:
It doesn't use any sleeping yet at all. Tokio itself does, but the current miri hook implementation just "wakes up early" (immediately), and then tokio tries to sleep again until the specified time is over. So we're busy waiting until the timer is done.

Reply:
Ah, so make this part of my complaint around incomplete shim implementations that dont document/check their limitations. ;)